src/settings/components/Email.tsx

```diff
@@ -54,22 +58,43 @@ const newsletterPreferencesReducer = (
 };
 
 const Email: FunctionComponent<EmailProps> = ({ user }) => {
+	const [t] = useTranslation();
+
+	const [isLoading, setIsLoading] = useState<boolean>(true);
 	const [initialNewsletterPreferences, setInitialNewsletterPreferences] = useState<
 		NewsletterPreferences
 	>(INITIAL_NEWSLETTER_PREFERENCES_STATE());
 	const [newsletterPreferences, changeNewsletterPreferences] = useReducer<
 		Reducer<NewsletterPreferences, NewsletterPreferencesAction>
 	>(newsletterPreferencesReducer, INITIAL_NEWSLETTER_PREFERENCES_STATE());
 
+	const GET_NEWSLETTER_LABELS = () => ({
+		newsletter: t(
+			'Ik ontvang graag tips en inspiratie voor mijn lessen en nieuws van partners.'
+		),
+		workshop: t('Ik wil berichten over workshops en events ontvangen.'),
+		ambassador: t(
+			'Ik krijg graag berichten om actief mee te werken aan Het Archief voor Onderwijs.'
+		),
+	});
+
+	const newsletterLabels = GET_NEWSLETTER_LABELS();
+
 	useEffect(() => {
-		fetchNewsletterPreferences(user.mail).then((preferences: NewsletterPreferences) => {
-			setInitialNewsletterPreferences(preferences);
-			changeNewsletterPreferences({
-				type: NewsletterPreferencesActionType.SET_NEWSLETTER_PREFERENCES,
-				payload: preferences,
+		try {
+			fetchNewsletterPreferences(user.mail).then((preferences: NewsletterPreferences) => {
+				setInitialNewsletterPreferences(preferences);
+				changeNewsletterPreferences({
+					type: NewsletterPreferencesActionType.SET_NEWSLETTER_PREFERENCES,
+					payload: preferences,
+				});
+				setIsLoading(false);
 			});
-		});
-	}, [user.mail]);
+		} catch (err) {
+			console.error(new CustomError('Failed to retrieve newsletter preferences', err));
+			ToastService.danger(t('De nieuwsbriefvoorkeuren konden niet worden opgevraagd.'));
+		}
```

Comment:
try catch werkt enkel bij await, niet als je een .then gebruikt.

Je hebt dus 2 opties.
* ofwel verzet je deze functie naar een useCallback functie, en roep je die aan in de useEffects
* ofwel gebruik je de `.catch()` functie ipv de `} catch() {`

ik ben persoonlijk voorstander van de eerste optie aangezien je dan een handige functie hebt als je bijvoorbeeld de preferences opnieuw wil fetchen:
```typescript
	const fetchEmailPreferences = useCallback(async () => {
		try {
			const preferences: NewsletterPreferences = await fetchNewsletterPreferences(user.mail);
			setInitialNewsletterPreferences(preferences);
			changeNewsletterPreferences({
				type: NewsletterPreferencesActionType.SET_NEWSLETTER_PREFERENCES,
				payload: preferences,
			});
		} catch (err) {
			console.error(new CustomError('Failed to retrieve newsletter preferences', err));
			ToastService.danger(t('De nieuwsbriefvoorkeuren konden niet worden opgevraagd.'));
		}
		setIsLoading(false);
	}, [user.mail, t]);

	useEffect(() => {
		fetchEmailPreferences();
	}, [fetchEmailPreferences]);
```

de `setLoading(false)` zet je altijd best na de catch, zodat de spinner ook stops als er zich een error voordoet.

Reply:
Thanks :)